src/config/request.rs

```diff
@@ -201,20 +214,14 @@ impl SetOpt for RequestConfig {
             easy.max_recv_speed(max)?;
         }
 
-        if let Some(cert) = self.ssl_client_certificate.as_ref() {
-            cert.set_opt(easy)?;
-        }
-
-        if let Some(cert) = self.ssl_ca_certificate.as_ref() {
-            cert.set_opt(easy)?;
-        }
-
-        if let Some(ciphers) = self.ssl_ciphers.as_ref() {
-            ciphers.set_opt(easy)?;
+        #[cfg(feature = "tls")]
+        if let Some(config) = self.tls_config.as_ref() {
+            config.set_opt_proxy(easy)?;
         }
 
-        if let Some(options) = self.ssl_options.as_ref() {
-            options.set_opt(easy)?;
+        #[cfg(feature = "tls")]
+        if let Some(config) = self.proxy_tls_config.as_ref() {
+            config.set_opt(easy)?;
```

Comment:
The bodies of the two if statements behave in the opposite way; `config.set_opt_proxy` should be set in the first and `config.set_opt` in the second statement.